parser-typechecker/src/Unison/Server/Endpoints/Projects.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Unison.Server.Endpoints.Projects where
+
+import Control.Error (runExceptT)
+import Control.Error.Util ((??))
+import Data.Aeson
+import Data.OpenApi (ToSchema)
+import qualified Data.Text as Text
+import Servant (QueryParam, throwError, (:>))
+import Servant.Docs (ToSample (..))
+import Servant.OpenApi ()
+import Servant.Server (Handler)
+import Unison.Codebase (Codebase)
+import qualified Unison.Codebase as Codebase
+import qualified Unison.Codebase.Branch as Branch
+import qualified Unison.Codebase.Path as Path
+import qualified Unison.Codebase.Path.Parse as Path
+import Unison.Codebase.ShortBranchHash (ShortBranchHash)
+import qualified Unison.Codebase.ShortBranchHash as SBH
+import qualified Unison.NameSegment as NameSegment
+import Unison.Parser.Ann (Ann)
+import Unison.Prelude
+import qualified Unison.Server.Backend as Backend
+import Unison.Server.Errors (backendError, badNamespace, rootBranchError)
+import Unison.Server.Types (APIGet, APIHeaders, UnisonHash, addHeaders)
+import Unison.Var (Var)
+
+type ProjectsAPI =
+  "projects" :> QueryParam "rootBranch" ShortBranchHash
+    :> APIGet [ProjectListing]
+
+instance ToSample ProjectListing where
+  toSamples _ =
+    [ ( "Projects in the root branch",
+        ProjectListing
+          (ProjectOwner "unison")
+          "base"
+          "#gjlk0dna8dongct6lsd19d1o9hi5n642t8jttga5e81e91fviqjdffem0tlddj7ahodjo5"
+      )
+    ]
+
+data ProjectOwner = ProjectOwner Text deriving (Generic, Show)
```

Comment:
Generally we want newtype for single-field datatypes unless we've got a good reason otherwise. It provides better performance characteristics. There are a few subtle behavioural changes, but they don't come up in 99% of cases.

```suggestion
newtype ProjectOwner = ProjectOwner Text deriving (Generic, Show)
```

Reply:
I did this initially (the LSP suggested it to me), but had some issues deriving the `ToJson` instance. I might bug you for some help on that, if I can't get it working.